*** tabledrag.orig.js	2010-06-04 11:59:17.000000000 +0100
--- tabledrag.js	2010-06-04 12:03:27.000000000 +0100
***************
*** 657,663 ****
        // Use the first row in the table as source, because it's garanteed to
        // be at the root level. Find the first item, then compare this row
        // against it as a sibling.
!       sourceRow = $('tr.draggable:first').get(0);
        if (sourceRow == this.rowObject.element) {
          sourceRow = $(this.rowObject.group[this.rowObject.group.length - 1]).next('tr.draggable').get(0);
        }
--- 657,663 ----
        // Use the first row in the table as source, because it's garanteed to
        // be at the root level. Find the first item, then compare this row
        // against it as a sibling.
!       sourceRow = $(this.table).find('tr.draggable:first').get(0);
        if (sourceRow == this.rowObject.element) {
          sourceRow = $(this.rowObject.group[this.rowObject.group.length - 1]).next('tr.draggable').get(0);
        }
