--- node.module.5.2.orig	2007-07-30 19:16:00.000000000 -0700
+++ node.module	2007-08-30 10:36:46.000000000 -0700
@@ -2718,6 +2718,23 @@ function node_access($op, $node = NULL) 
     return $access;
   }
 
+/**
+ * Modification
+ * Nodeapi Access: http://drupal.org/node/143075, http://drupal.org/node/122173
+ * This is for TAC/OG Integration. http://groups.drupal.org/node/4026
+ * Check nodeapi 'access' hook.
+ * This happens before the module's own hook so that nodeapi can perform
+ * overrides.
+ */
+  $access = node_invoke_nodeapi($node, 'access', $op);
+  // node_invoke_nodeapi drops NULLs; so if no nodeapi actually responded
+  // to this, we'll have an empty return value, which means nothing cares
+  // and we move on.
+  if (!empty($access) && is_array($access)) {
+    // If any values were true, return true; otherwise, return false.
+    return (bool) array_filter($access);
+  }
+
   // If the module did not override the access rights, use those set in the
   // node_access table.
   if ($op != 'create' && $node->nid && $node->status) {
