--- og.module.5.x-3.x-dev.orig	2007-09-22 23:12:36.000000000 -0700
+++ og.module	2007-09-22 23:12:45.000000000 -0700
@@ -2277,7 +2277,8 @@ function og_og_block_details($node) {
   }
 
   $block['content'] = theme('item_list', $links). $post;
-  $block['subject'] = $node->title;
+//  $block['subject'] = $node->title;
+  $block['subject'] = l("$node->title", "node/$node->nid");
   return $block;
 }
 
