+++ (current document)
@@ -1245,10 +1245,6 @@
 $view->tid = view_controls_get_view_tid($view);
 
 	// add content link
-	
-	//patch to work with scratchpad permissions
-	$ispecies = ispecies_ispecies();
-	if(!$ispecies[$view->name]){
 
 		// are there more than one type of node for this view?
 		foreach($view->filter as $filter){
@@ -1308,8 +1304,6 @@
 		$view_controls = l($img, $add_link, $attributes, NULL, $fragment, FALSE, TRUE);
 
 		}
-
-	}	
 	
 	// configuration link